### PR TITLE
fix: decouple color scheme storage from design system

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Why
- Currently design system is using some color scheme hooks. The color scheme hook is using listener from mmkv leading to mmkv imports https://github.com/showtime-xyz/showtime-frontend/issues/1179
- To separate mmkv/localstorage implementation from design system.
- In future, we'll remove the color scheme and theme from components but for now we can just separate color scheme.
- To simplify some complexity in color mode implementation and clean up some code!

# How
- Created a `ColorSchemeProvider` in design system. We need to publish it under `@showtime-xyz/universal.color-scheme-provider`.
- Refer [ThemeProvider](https://github.com/showtime-xyz/showtime-frontend/compare/fix/decouple-color-scheme?expand=1#diff-7d406e7647cfaf3ed7efddb63ff246d56ac7176b4cdabbffb448bb6c9b8a96b1L18) for usage of this provider.

# Test Plan
- I have locally tested on web and native by aliasing @showtime-xyz/universal to design-system.
- We need to publish some components that are dependent upon `useColorScheme` hook.

P.S. To test it completely we need to publish some components. So putting this PR in review.

😌 
Screenshot 2022-06-16 at 6 42 57 PM

### CODE REVIEW

**Comment by alantoa** [comment]
maybe this PR will fix this issue: 

image

**Comment by intergalacticspacehighway** [comment]
> maybe this PR will fix this issue:

yeah it should. I think we should release all the components, merge this PR and wait for tailwindcss-react-native to have a next stable release and test it later. wdyt?

**Comment by alantoa** [comment]
> yeah it should. I think we should release all the components, merge this PR and wait for tailwindcss-react-native to have a next stable release and test it later. wdyt?

yeah, merge this PR is well!
but upgrading all the components now has some issues, I'm digging into. you can see this [branch](https://github.com/showtime-xyz/showtime-frontend/tree/feature/show2-875-design-system-compoents), `bit` seems to have some problems.

**Comment by alantoa** [comment]
hi, @intergalacticspacehighway!
vercel is built fail, you can release `@showtime-xyz/universal.color-scheme-provider`  component alone?

File: .bitmap [hunk 1]
```diff
@@ -11,253 +11,259 @@
...
     "checkbox": {
         "scope": "showtime.universal",
-        "version": "0.0.8",
+        "version": "0.0.12",
         "mainFile": "index.tsx",
         "rootDir": "packages/design-system/checkbox"
     },
+    "color-scheme": {
+        "scope": "",
+        "version": "",
```
Comment (axeldelafosse) [inline .bitmap]: Looks like something missing here

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

File: .bitmap (hunk 1 above)
Comment (alantoa) [reply .bitmap]: since it hasn't been released yet, trying.

Comment (axeldelafosse) [reply .bitmap]: Also, are you trying to use `@showtime-xyz/universal.color-scheme` or `@showtime-xyz/universal.color-scheme-provider`?

Comment (intergalacticspacehighway) [reply .bitmap]: yeah i had renamed @showtime-xyz/universal.color-scheme-provider to @showtime-xyz/universal.color-scheme
sorry. naming OCD kicked in

**Review by alantoa — APPROVED** [review]
LFG! ✨

File: .bitmap (hunk 1 above)
Comment (axeldelafosse) [reply .bitmap]: > naming OCD kicked in

🤣

**Pull request merged**